src/Configuration/GeneratorConfiguration.php

```diff
@@ -36,16 +36,23 @@ class GeneratorConfiguration implements \IteratorAggregate
      */
     private $classesToGenerate = [];
 
-    public function __construct(array $defaultGroupCombinations, array $defaultVersions)
+    /**
+     * @var bool
+     */
+    private $assignUnknownArrays;
+
+    public function __construct(array $defaultGroupCombinations, array $defaultVersions, bool $assignUnknownArrays = false)
```

Comment:
i would prefer to introduce a variable `array $options = []` here. otherwise the constructor will become cumbersome if we add more flags. currently the only supported flag will be `assign_unknown_arrays`.

we can require symfony/options-resolver (it is indirectly required anyways already) and use that to resolve the options.

Reply:
You are right, an options array is probably better, implemented via fe9e86955755d17308d58712b4f9a011ce97a563